addons/purchase/views/purchase_views.xml

```diff
@@ -800,11 +800,20 @@
                     <field name="product_id"/>
                     <field name="partner_id" string="Vendor"/>
                     <filter name="hide_cancelled" string="Hide cancelled lines" domain="[('state', '!=', 'cancel')]"/>
+                    <filter string="Status" name="purchase_state" domain="[('state', '=', 'purchase')]" />
+                    <separator/>
+                    <filter name="filter_date" date="date_order" default_period="month"/>
```

Comment:
```suggestion
                    <filter name="filter_date_order" date="date_order" default_period="month"/>
```
In conclusion
filter and group_by name should be `{filter/group_by}_{field_name}` for date and if filter is doing something like getting last_month date then `filter_last_month`

I hope you understand it :pray:

Reply:
Got the point :+1: 
Thankyou